### ISSUE TITLE
Site Editor should display a 404 message when you navigate to a non-existing path

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#67445]
### Description

With a block theme and Gutenberg trunk active, if you open the Site Editor but miss-type the link, the Site Editor is now a black page with no message or content showing.

Without Gutenberg active, you are presented with the index template (Or home blog or front page, if the theme uses that)

### Step-by-step reproduction instructions

Try navigating to  `wp-admin/site-editor.php?p=%2Ftemplate-fdgfdgdfg`.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by carolinan** [comment]
@youknowriad I believe this is related to https://github.com/WordPress/gutenberg/pull/67199.

**Comment by youknowriad** [comment]
Yes, we discussed this in the PR. We should define a real 404 page instead of showing the home page like before.

**Comment by carolinan** [comment]
💯 
Is there another issue already opened for that?

**Comment by youknowriad** [comment]
No, we can use this one I think.
